AIRTools/GoogleServices.cs

```diff
@@ -25,11 +28,11 @@ public static async Task FromJson(string path, bool addToAne = true)
             var storageBucket = projectInfo["storage_bucket"]?.ToString();
             var client = JArray.FromObject(googleServices["client"]).First;
             var apiKey = JArray.FromObject(client?["api_key"]!).First;
-            var oauthClient = JArray.FromObject(client?["oauth_client"]!).First;
             var clientInfo = JObject.FromObject(client?["client_info"]!);
             var mobilesdkAppId = JObject.FromObject(clientInfo!)["mobilesdk_app_id"]?.ToString();
             var currentKey = JObject.FromObject(apiKey!)["current_key"]?.ToString();
-            var clientId = JObject.FromObject(oauthClient!)["client_id"]?.ToString();
+            var webClient = client?["oauth_client"]!.First(c => (int) c["client_type"] == WebClientType);
```

Comment:
won't the First here `client?["oauth_client"]!.First` return the first element in the array?

So if the first element is "client_type": 4 then this will fail?

"oauth_client": [
        {
          "client_id": "xxxx.apps.googleusercontent.com",
          "client_type": 3
        },
        {
          "client_id": "xxxx.apps.googleusercontent.com",
          "client_type": 4
        }
      ],

Feel like there should be a cleaner way to extract the object by filtering the array instead

Reply:
LINQ `.First(predicate)` will return the first element matching `predicate`, it's like shortcut for `.Where(predicate).First()`